src/sailfishusermanager.cpp

```diff
@@ -358,6 +360,25 @@ uint SailfishUserManager::addSailfishUser(const QString &user, const QString &na
         return 0;
     }
 
+    // Execute user creation scripts
+    QDir creation(USER_CREATE_SCRIPT_DIR);
+    creation.setSorting(QDir::NoSort);
+    creation.setFilter(QDir::Files | QDir::Executable);
+
+    auto entryList = creation.entryList();
```

Comment:
You could have just `creation.entryList(QStringList() << "*.sh", QDir::Files | QDir::Executable, QDir::Name)`. That sorts by name. Just like in `removeUserFiles()`.

Reply:
QDir::Name will sort like this 
1.sh
10.sh
2.sh

instead 
1.sh
2.sh
10.sh

Reply:
Yes. Then again it's pretty much a convention to name such files like `01-foo.sh`, `02-bar.sh` and `10-foobar.sh` which maintains the expected order even with `QDir::Name`. Currently none of the files we have is named with a number at the beginning though.

What actually matters here is that the order is deterministic, i.e. it doesn't change from device to device so the behaviour is always the same. Without any sorting the order depeds on for example the order the packages where installed and if there is some dependency between the files, then you any bugs that occure might not be easily reproducible. It's less important to ensure that the order is natural than to ensure that it's always the same.